Merge node sections on production deployment page (#832)
redpanda-data/documentation: modules/reference/pages/rpk/rpk-transform/rpk-transform-deploy.adoc

|===
|*Value* |*Type* |*Description*

|--compression |string |Output batch compression type.

|--file |string |The WebAssembly module to deploy.

> |--from-offset |string |Process an input topic partition from a relative offset; check help text for more information.

Reviewer: ```suggestion
|--from-offset |string |Process an input topic partition from a relative offset. Run `rpk transform deploy --from-offset -h` for more information.

```

|===
|*Value* |*Type* |*Description*

|--compression |string |Output batch compression type.

|--file |string |The WebAssembly module to deploy.

> |--from-offset |string |Process an input topic partition from a relative offset. Run `rpk transform deploy --from-offset -h` for more information.

Reviewer: ```suggestion
|--from-offset |string |Process an input topic partition from a relative offset.
```

|--file |string |The WebAssembly module to deploy.

> |--from-offset |string |Process an input topic partition from a relative offset. Run `rpk transform deploy --from-offset -h` for more information.

Reviewer 2: if this is the formulation we want, then it should be

```suggestion
|--from-offset |string |Process an input topic partition from a relative offset. Run `rpk transform deploy -h` for more information.
```

rpk transform deploy --var FOO=BAR --var FIZZ=BUZZ
----

> Configure compression for batches output by data transforms. The default setting is "no compression," but you can choose from the following options:

Reviewer: ```suggestion
Configure compression for batches output by data transforms. The default setting is `none` but you can choose from the following options:

```

rpk transform deploy --var FOO=BAR --var FIZZ=BUZZ
----

> Configure compression for batches output by data transforms. The default setting is "no compression," but you can choose from the following options:

Reviewer 3: Is "no compression" the same as `none` listed below?

Reviewer: Yes. I'll update

* none
* gzip
* snappy
* lz4
* zstd

Configure this at deployment using `rpk` with the `--compression` flag:

[,bash]
----
rpk transform deploy --compression <compression_type>
----

Enabling compression may increase computation costs and could impact latency at the output topic.

> For more details, see xref:modules:deploy/deployment-option/self-hosted/manual/sizing.adoc[].

Reviewer 3: ```suggestion
For more details, see xref:deploy:deployment-option/self-hosted/manual/sizing.adoc[].
```
Please double check - maybe ROOT needs to be added?

Reviewer: That fix is correct. Thanks Kat
